### DISCUSSION TITLE
Why doesn't Fairlearn specify which group is privileged?

### DISCUSSION THREAD

**Discussion by fountaindive** [fairlearn/fairlearn#1319] [Q&A]
I was wondering what the motivation is for not specifying a particular group as 'privileged' or 'unprivileged' in a calculation. For example, in the way ai-fairness-360 does.

In Fairlearn I noticed that for instance when computing differences you compute the absolute differences. I like this in that it makes it easier to use but wouldn't you still like to the direction of the disparity?

Thanks!!

**Comment by romanlutz** [comment]
Just speaking for myself here, but that's a part of AIF360 I really dislike. MetricFrame or other utilities in the metrics module give you all the numbers or even ratio and difference. Why do I need this additionally?

The word privileged carries a lot of meaning in an application context. So if I'm just exploring the data without a deep understanding yet, how can I decide to label a group a privileged? It feels like I'm making choices that I am not really educated to make at an early stage. Sociotechnical contexts tend to be less binary (privileged, unprivileged) and a lot more complex, too. So without really losing any functionality, I would very much prefer to stay away from such terms in an API.

If it's just meant to mean "the group with highest value of metric M" then why call it privileged? Just call it max.

But again, just my 2 cents here. Other @fairlearn/fairlearn-maintainers may have different perspectives.

**Reply by fountaindive** [reply]
Hi @romanlutz,

Thanks for your thoughts. I also don't really like AIF360 terminology on this either. When computing some disparity measure, for example demographic parity difference between two groups A and B, I find it better to interpret the results simply as "group A has higher probability of favourable outcome" or "group B has higher probability of favourable outcome" depending on the sign of the result.

That brings me back to my other question I asked actually. Can you tell me why fairlearn computes absolute differences and ratios \in [0,1]?

It would be nice to be able to keep track of which groups in a sensitive feature have higher or lower relative scores?

Thanks!

**Reply by romanlutz** [reply]
Maybe I'm not thinking about the scenario you're working on, but we are not specifying which groups to compare at all, so the formula is simply max-min which is non-negative by definition. Same for ration with min/max.

If you want to know which groups were involved just get by_group of your MetricFrame and it'll show you. 

I don't really see the scenario where we should have a function for calculating a simple difference of two groups that you name, because then you can do it yourself just as easily without a helper function from fairlearn. But again, I'm possibly ignoring something valid that doesn't occur to me without example.

**Reply by fountaindive** [reply]
Ahhh ok that makes sense now with the max-min approach thanks!

I wasn't really asking for the ability to compute the difference between two groups but rather have the ability to know which two groups were used in the `mf.difference()` calculation. Maybe it's just how I've set up my dataframes but when I do something like
`mf.group_min()` it returns the a pandas.Series of the minimum value for each metric but it doesn't say which groups they came from.

So let's say that one of the metrics indicated that there was some unfairness and your sensitive feature has N categories. You wouldn't know which two specific categories were being compared. But I'm still a novice with fairlearn so I apologise if you can get this information!

Either way, it's not really a big deal because it's relatively straight forward to get this information out and I want to look at the distribution of e.g. the TPRs for each category in a sensitive feature anyway.

**Reply by romanlutz** [reply]
I think I know what you mean. Even if I have the following (taken from user guide)
```
>>> mf.by_group
                     tpr       fpr   sel  count
sensitive_feature_0
a                    0.5  1.000000  0.75    4.0
b                    0.6  0.000000  0.50    6.0
c                    0.4  0.666667  0.50    8.0
```

and if there are many groups I'd have to look through to find the smallest value and largest value, whereas you're looking for something that doesn't just tell me 
```
mf.group_min()
tpr      0.4
fpr      0.0
sel      0.5
count    4.0
dtype: object
```
but, say, something like
```
mf.group_min()
tpr      0.4     c
fpr      0.0     b
sel      0.5     b,c
count    4.0     a
dtype: object
```
Please confirm if I got it this time 😄 If so, we can perhaps think about ways to incorporate that, or at least how to add a complementary argmin/argmax.

**Reply by fountaindive** [reply]
Sorry, this replay got totally lost in my email!

Yeah something like that so you can more easily find out which groups were involved in the calculation would be useful though I think.
